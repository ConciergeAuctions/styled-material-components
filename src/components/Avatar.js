// @flow
import React, { type Node } from "react";

import { Circular } from "./Circular";
import { Image } from "./Image";

export type AvatarPropsType = {|
  children: Node,
  name?: string,
  src?: string
|};

const formatProps = ({
  name,
  children
}: {
  children: React.Node,
  name?: string
}): string | React.Node => {
  if (!name) {
    return children;
  } else if (name.length <= 2) {
    return name;
  }

  const nameArr = name.split(" ");
  const initials =
    nameArr.length === 1 ? nameArr[0][0] : nameArr[0][0] + nameArr[1][0];

  return initials.toUpperCase();
};

export const Avatar = Circular.extend.attrs({
  // eslint-disable-next-line react/display-name
  children: (props: AvatarPropsType): string | React.Node =>
    props.src ? <Image src={props.src} /> : formatProps(props)
})`
  // This is just here because we have to have some sort of styles applied at this level.
  text-align: left;
`;
