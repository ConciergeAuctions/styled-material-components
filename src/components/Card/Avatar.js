import React from 'react';
import styled from 'styled-components';

const AvatarComponent = props => (
  <div className={`${props.className} smc-avatar`}>{props.children}</div>
);

const Avatar = styled(AvatarComponent)`
  position: absolute;
  background: #bdbdbd;
  height: 2.5rem;
  width: 2.5rem;
  border-radius: 50%;
`;

export default Avatar;
