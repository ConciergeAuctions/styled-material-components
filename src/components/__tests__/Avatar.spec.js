import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import {
  STYLED_COMPONENTS_CONTEXT,
  STYLED_COMPONENTS_CHILD_CONTEXT_TYPES,
} from '../../../test/utils';

import { Avatar } from '../Avatar';

describe('Avatar with src prop', () => {
  let mounted;
  it('mounts', () => {
    const mounted = shallow(<Avatar src="test.jpg" />, {
      context: { ...STYLED_COMPONENTS_CONTEXT },
      childContextTypes: { ...STYLED_COMPONENTS_CHILD_CONTEXT_TYPES },
    });
    expect(mounted).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(toJSON(mounted)).toMatchSnapshot();
  });

  it.skip('renders an image with the src provided', () => {});
});
