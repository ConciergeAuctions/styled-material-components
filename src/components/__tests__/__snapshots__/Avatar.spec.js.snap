// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar with src prop matches snapshot 1`] = `undefined`;
